src/components/services-view/services-view.controller.ts

```diff
@@ -3,35 +3,61 @@ import * as _ from 'lodash';
 import * as $ from 'jquery';
 
 export class ServicesViewController implements angular.IController {
-  static $inject = ['Constants', 'Catalog', 'Logger', 'HTMLService', '$filter', '$scope', '$timeout'];
+  static $inject = ['Constants', 'Catalog', 'KeywordService', 'Logger', 'HTMLService', '$filter', '$rootScope', '$scope', '$timeout'];
 
   public ctrl: any = this;
   private constants: any;
   private catalog: any;
+  private keywordService: any;
   private logger: any;
   private htmlService: any;
   private $filter: any;
+  private $rootScope: any;
   private $scope: any;
   private $timeout: any;
   private debounceResize: any;
 
-  constructor(constants: any, catalog: any, logger: any, htmlService: any, $filter: any, $scope: any, $timeout: any) {
+  constructor(constants: any, catalog: any, keywordService: any, logger: any, htmlService: any, $filter: any, $rootScope: any, $scope: any, $timeout: any) {
     this.constants = constants;
     this.catalog = catalog;
+    this.keywordService = keywordService;
     this.logger = logger;
     this.htmlService = htmlService;
     this.$filter = $filter;
+    this.$rootScope = $rootScope;
     this.$scope = $scope;
     this.$timeout = $timeout;
     this.ctrl.loaded = false;
     this.ctrl.isEmpty = false;
     this.ctrl.mobileView = 'categories';
+    this.ctrl.filterConfig = {};
   }
 
   public $onInit() {
     this.debounceResize = _.debounce(this.resizeExpansion, 50, { maxWait: 250 });
     angular.element(window).bind('resize', this.debounceResize);
     $(window).on('resize.services', this.debounceResize);
+
+    this.$rootScope.$on('filter-catalog-items', (event: any, searchCriteria: any) => {
+      let keyWordFilter: any = angular.copy(this.ctrl.filterConfig.fields[0]);
```

Comment:
i would prefer to have named references vs array indexing for things like this, create a named reference to the field, and then when you setup ctrl.filterConfig fields just include the field via the ref there, and then use the ref here as well